teamcity: failed tests on master: testrace (proposer evaluated kv)/TestLeaseExtensionNotBlockedByRead

The following tests appear to have failed:

[#99506](https://teamcity.cockroachdb.com/viewLog.html?buildId=99506):

```
--- FAIL: testrace (proposer evaluated kv)/TestLeaseExtensionNotBlockedByRead (0.012s)
Race detected!
W161224 01:22:43.500621 29253 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I161224 01:22:43.503191 29253 server/config.go:447  1 storage engine initialized
I161224 01:22:43.504416 29253 server/node.go:426  [n?] store [n0,s0] not bootstrapped
I161224 01:22:43.517449 29253 server/node.go:355  [n?] **** cluster 283f8f14-e919-4b7c-b3d3-cd849ade7b8d has been created
I161224 01:22:43.517557 29253 server/node.go:356  [n?] **** add additional nodes by specifying --join=127.0.0.1:42393
I161224 01:22:43.529566 29253 storage/store.go:1258  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161224 01:22:43.530286 29253 server/node.go:439  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0}
I161224 01:22:43.538803 29253 server/node.go:324  [n1] node ID 1 initialized
I161224 01:22:43.539005 29253 gossip/gossip.go:292  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42393" > attrs:<> locality:<> 
I161224 01:22:43.539445 29253 storage/stores.go:296  [n1] read 0 node addresses from persistent storage
I161224 01:22:43.539666 29253 server/node.go:569  [n1] connecting to gossip network to verify cluster ID...
I161224 01:22:43.539792 29253 server/node.go:589  [n1] node connected via gossip and verified as part of cluster "283f8f14-e919-4b7c-b3d3-cd849ade7b8d"
I161224 01:22:43.540162 29253 server/node.go:374  [n1] node=1: started with [[]=] engine(s) and attributes []
I161224 01:22:43.540578 29253 sql/executor.go:322  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:42393}
I161224 01:22:43.554313 29230 storage/split_queue.go:101  [split,n1,s1,r1/1:/M{in-ax},@c4200b4000] splitting at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I161224 01:22:43.565992 29230 storage/replica_command.go:2359  [split,n1,s1,r1/1:/M{in-ax},@c4200b4000] initiating a split of this range at key /Table/11 [r2]
I161224 01:22:43.570593 29253 server/server.go:621  [n1] starting https server at 127.0.0.1:53522
I161224 01:22:43.570694 29253 server/server.go:622  [n1] starting grpc/postgres server at 127.0.0.1:42393
I161224 01:22:43.570772 29253 server/server.go:623  [n1] advertising CockroachDB node at 127.0.0.1:42393
I161224 01:22:43.628169 29275 sql/event_log.go:95  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:42393} Attrs: Locality:} ClusterID:283f8f14-e919-4b7c-b3d3-cd849ade7b8d StartedAt:1482542563539820977}
E161224 01:22:43.666323 29230 storage/queue.go:597  [split,n1,s1,r1/1:/{Min-Table/11},@c4200b4000] unable to split [n1,s1,r1/1:/{Min-Table/11}] at key "/Table/12/0": key range /Table/12/0-/Table/12/0 outside of bounds of range /Min-/Max
I161224 01:22:43.667869 29230 storage/split_queue.go:101  [split,n1,s1,r2/1:/{Table/11-Max},@c42020e000] splitting at keys [/Table/12/0 /Table/13/0 /Table/14/0]
I161224 01:22:43.668120 29230 storage/replica_command.go:2359  [split,n1,s1,r2/1:/{Table/11-Max},@c42020e000] initiating a split of this range at key /Table/12 [r3]
I161224 01:22:43.694184 29253 sql/event_log.go:95  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN uniqueID SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
E161224 01:22:43.704383 29230 storage/queue.go:597  [split,n1,s1,r2/1:/Table/1{1-2},@c42020e000] unable to split [n1,s1,r2/1:/Table/1{1-2}] at key "/Table/13/0": key range /Table/13/0-/Table/13/0 outside of bounds of range /Table/11-/Max
I161224 01:22:43.705007 29230 storage/split_queue.go:101  [split,n1,s1,r3/1:/{Table/12-Max},@c4202b3b00] splitting at keys [/Table/13/0 /Table/14/0]
I161224 01:22:43.705222 29230 storage/replica_command.go:2359  [split,n1,s1,r3/1:/{Table/12-Max},@c4202b3b00] initiating a split of this range at key /Table/13 [r4]
E161224 01:22:43.753915 29230 storage/queue.go:597  [split,n1,s1,r3/1:/Table/1{2-3},@c4202b3b00] unable to split [n1,s1,r3/1:/Table/1{2-3}] at key "/Table/14/0": key range /Table/14/0-/Table/14/0 outside of bounds of range /Table/12-/Max
I161224 01:22:43.755416 29230 storage/split_queue.go:101  [split,n1,s1,r4/1:/{Table/13-Max},@c420e16000] splitting at keys [/Table/14/0]
I161224 01:22:43.755622 29230 storage/replica_command.go:2359  [split,n1,s1,r4/1:/{Table/13-Max},@c420e16000] initiating a split of this range at key /Table/14 [r5]
I161224 01:22:43.805015 29253 server/server.go:676  [n1] done ensuring all necessary migrations have run
I161224 01:22:43.805185 29253 server/server.go:678  [n1] serving sql connections
I161224 01:22:43.877470 29253 util/stop/stopper.go:472  quiescing; tasks left:
1      server/node.go:836
==================
WARNING: DATA RACE
Read at 0x00c421959080 by goroutine 411:
  reflect.Value.Int()
      /usr/local/go/src/reflect/value.go:888 +0x15b
  encoding/gob.encInt()
      /usr/local/go/src/encoding/gob/encode.go:188 +0x50
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:357 +0x2c9
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:557 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:707 +0x257
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:247 +0x6ea
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/pkg/kv.grpcTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:83 +0x44d
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Previous write at 0x00c421959080 by goroutine 232:
  [failed to restore the stack]

Goroutine 411 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x73
  github.com/cockroachdb/cockroach/pkg/kv.grpcTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:108 +0x2d2
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1123 +0x1c3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:405 +0x3bf
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:316 +0x2f4
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:439 +0x553
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:323 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:263 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:433 +0x73e
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:323 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:263 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:443 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeIterator).Seek()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_iter.go:157 +0x3ed
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:720 +0x23a
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:620 +0x381
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:418 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:587 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:711 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:418 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0xfd
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:168 +0xee
  github.com/cockroachdb/cockroach/pkg/sql.getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/descriptor.go:202 +0xd6
  github.com/cockroachdb/cockroach/pkg/sql.getDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:142 +0x181
  github.com/cockroachdb/cockroach/pkg/sql.MustGetDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:159 +0x70
  github.com/cockroachdb/cockroach/pkg/sql.getTableOrViewDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:146 +0x190
  github.com/cockroachdb/cockroach/pkg/sql.getTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:170 +0x60
  github.com/cockroachdb/cockroach/pkg/sql.mustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:215 +0x66
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).mustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:209 +0x91
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getTableLease()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:275 +0xdbe
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeEditNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/update.go:44 +0x5d
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:60 +0xf8
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:298 +0x2f1e
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:217 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:296 +0x14c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:357 +0x9e
  github.com/cockroachdb/cockroach/pkg/sql.InternalExecutor.ExecuteStatementInTransaction()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:47 +0x163
  github.com/cockroachdb/cockroach/pkg/sql.EventLogger.InsertEventRecord()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/event_log.go:121 +0x4ff
  github.com/cockroachdb/cockroach/pkg/server.(*Node).recordJoinEvent.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:747 +0x1cd
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:468 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:520 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:469 +0x297
  github.com/cockroachdb/cockroach/pkg/server.(*Node).recordJoinEvent.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:748 +0x399
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Goroutine 232 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:206 +0x225
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1136 +0x432
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:405 +0x3bf
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:474 +0x1aa
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:926 +0x3b3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:805 +0x567
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:620 +0x381
  github.com/cockroachdb/cockroach/pkg/internal/client.SendWrappedWith()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:53 +0x174
  github.com/cockroachdb/cockroach/pkg/internal/client.SendWrapped()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:70 +0xbf
  github.com/cockroachdb/cockroach/pkg/storage_test.TestLeaseExtensionNotBlockedByRead()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_replica_test.go:894 +0x951
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
==================
==================
WARNING: DATA RACE
Read at 0x00c421959088 by goroutine 411:
  reflect.Value.Int()
      /usr/local/go/src/reflect/value.go:886 +0x192
  encoding/gob.encInt()
      /usr/local/go/src/encoding/gob/encode.go:188 +0x50
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:357 +0x2c9
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:557 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:707 +0x257
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:247 +0x6ea
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/pkg/kv.grpcTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:83 +0x44d
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Previous write at 0x00c421959088 by goroutine 232:
  [failed to restore the stack]

Goroutine 411 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x73
  github.com/cockroachdb/cockroach/pkg/kv.grpcTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:108 +0x2d2
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1123 +0x1c3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:405 +0x3bf
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:316 +0x2f4
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:439 +0x553
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:323 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:263 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:433 +0x73e
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:323 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:263 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:443 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeIterator).Seek()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_iter.go:157 +0x3ed
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:720 +0x23a
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:620 +0x381
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:418 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:587 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:711 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:418 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0xfd
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:168 +0xee
  github.com/cockroachdb/cockroach/pkg/sql.getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/descriptor.go:202 +0xd6
  github.com/cockroachdb/cockroach/pkg/sql.getDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:142 +0x181
  github.com/cockroachdb/cockroach/pkg/sql.MustGetDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:159 +0x70
  github.com/cockroachdb/cockroach/pkg/sql.getTableOrViewDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:146 +0x190
  github.com/cockroachdb/cockroach/pkg/sql.getTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:170 +0x60
  github.com/cockroachdb/cockroach/pkg/sql.mustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:215 +0x66
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).mustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:209 +0x91
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getTableLease()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:275 +0xdbe
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeEditNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/update.go:44 +0x5d
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:60 +0xf8
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:298 +0x2f1e
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:217 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:296 +0x14c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:357 +0x9e
  github.com/cockroachdb/cockroach/pkg/sql.InternalExecutor.ExecuteStatementInTransaction()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:47 +0x163
  github.com/cockroachdb/cockroach/pkg/sql.EventLogger.InsertEventRecord()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/event_log.go:121 +0x4ff
  github.com/cockroachdb/cockroach/pkg/server.(*Node).recordJoinEvent.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:747 +0x1cd
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:468 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:520 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:469 +0x297
  github.com/cockroachdb/cockroach/pkg/server.(*Node).recordJoinEvent.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:748 +0x399
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Goroutine 232 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:206 +0x225
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1136 +0x432
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:405 +0x3bf
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:474 +0x1aa
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:926 +0x3b3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:805 +0x567
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:620 +0x381
  github.com/cockroachdb/cockroach/pkg/internal/client.SendWrappedWith()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:53 +0x174
  github.com/cockroachdb/cockroach/pkg/internal/client.SendWrapped()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/sender.go:70 +0xbf
  github.com/cockroachdb/cockroach/pkg/storage_test.TestLeaseExtensionNotBlockedByRead()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_replica_test.go:894 +0x951
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
==================
I161224 01:22:43.894045 29207 kv/transport_race.go:71  transport race promotion: ran 15 iterations on up to 122 requests

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -148,9 +148,12 @@ func (r *Replica) executeCmd(
 
 	if cmd, ok := commands[args.Method()]; ok {
 		cArgs := CommandArgs{
-			Repl:    r,
-			Header:  h,
-			Args:    args,
+			Repl:   r,
+			Header: h,
+			// Some commands mutate their arguments, so give each invocation
+			// its own copy (shallow to mimic earlier versions of this code
+			// in which args were passed by value instead of pointer).
+			Args:    args.ShallowCopy(),
 			MaxKeys: maxKeys,
 			Stats:   ms,
 		}

